src/Curation/Curation.service.ts

```diff
@@ -0,0 +1,116 @@
+import { raw, SQL } from 'decentraland-server'
+import { hasAccess as hasCollectionAccess } from '../Collection/access'
+import { getMergedCollection } from '../Collection/utils'
+import { hasAccess as hasItemAccess } from '../Item/access'
+import { getMergedItem } from '../Item/utils'
+import { CollectionCuration } from './CollectionCuration'
+import { CurationType } from './Curation.types'
+import { ItemCuration } from './ItemCuration'
+
+export class CurationService<
+  T extends typeof CollectionCuration | typeof ItemCuration
+> {
+  public type: CurationType
+
+  // TODO: Type this better (or remove it completely and use+mock new CurationService)
+  static byType(type: CurationType): CurationService<any> {
+    switch (type) {
+      case CurationType.COLLECTION:
+        return new CurationService(CollectionCuration)
+      case CurationType.ITEM:
+        return new CurationService(ItemCuration)
+      default:
+        throw new Error(`Invalid curation type ${type}`)
+    }
+  }
+
+  constructor(public CurationModel: T) {
+    switch (CurationModel) {
+      case CollectionCuration:
```

Comment:
This is comparing the class definition reference, right?

Reply:
Correct